packages/http-client-java/generator/http-client-generator-mgmt/src/main/java/com/microsoft/typespec/http/client/generator/mgmt/template/SampleTemplate.java

```diff
@@ -24,8 +25,9 @@ public String write(List<FluentExample> examples, List<JavaFile> sampleJavaFiles
         assert examples.size() == sampleJavaFiles.size();
 
         // clean up copyright etc.
-        Map<String, String> javaFiles
-            = sampleJavaFiles.stream().collect(Collectors.toMap(JavaFile::getFilePath, SampleTemplate::cleanJavaFile));
+        Map<String, String> javaFiles = sampleJavaFiles.stream()
+            .collect(Collectors.toMap(JavaFile::getFilePath, SampleTemplate::cleanJavaFile,
+                (existing, replacement) -> existing, LinkedHashMap::new));
```

Comment:
Is the order problem happens here in `collect`, or happens in `javaFileContents = CodeFormatterUtil.formatCode(javaFiles)`?

Reply:
in `collect`. 
`CodeFormatterUtil.formatCode(javaFiles)` won't change entry ordering.

Reply:
OK, so here the implicit assumption is that the entry order in Map should not be changed?